packages/ensrainbow-sdk/src/client.ts

```diff
@@ -208,44 +309,157 @@ export class EnsRainbowApiClient implements EnsRainbow.ApiClient {
+        errorMessage.toLowerCase().includes("failed to fetch")
+      ) {
+        return {
+          status: StatusCode.Error,
+          error: "Network connection lost or unavailable",
+          errorCode: ErrorCode.NETWORK_OFFLINE,
+        } as EnsRainbow.HealNetworkOfflineError;
+      }
+    }
+
+    // Default to general network error
+    return {
+      status: StatusCode.Error,
+      error: errorMessage,
+      errorCode: ErrorCode.GENERAL_NETWORK_ERROR,
+    } as EnsRainbow.HealGeneralNetworkError;
   }
 
   /**
    * Get Count of Healable Labels
    *
    * @returns a `CountResponse` indicating the result and the timestamp of the request and the number of healable labels if successful
-   * @throws if the request fails due to network failures, DNS lookup failures, request timeouts, CORS violations, or Invalid URLs
-   *
    * @example
    *
    * const response = await client.count();
    *
    * console.log(response);
    *
+   * // Success case:
    * // {
    * //   "status": "success",
    * //   "count": 133856894,
    * //   "timestamp": "2024-01-30T11:18:56Z"
    * // }
    *
+   * // Server error case:
+   * // {
+   * //   "status": "error",
+   * //   "error": "Server error",
+   * //   "errorCode": 500
+   * // }
+   *
+   * // Network timeout error case:
+   * // {
+   * //   "status": "error",
+   * //   "error": "Connection timed out",
+   * //   "errorCode": 1000
+   * // }
+   *
+   * // Network offline error case:
+   * // {
+   * //   "status": "error",
+   * //   "error": "Server is unreachable",
+   * //   "errorCode": 1001
+   * // }
+   *
+   * // General network error case:
+   * // {
+   * //   "status": "error",
+   * //   "error": "Unknown network error",
+   * //   "errorCode": 1099
+   * // }
    */
   async count(): Promise<EnsRainbow.CountResponse> {
-    const response = await fetch(new URL("/v1/labels/count", this.options.endpointUrl));
+    try {
+      return await this.fetchWithTimeout<EnsRainbow.CountResponse>("/v1/labels/count");
+    } catch (error) {
+      // Handle network errors
+      return this.createCountNetworkErrorResponse(error);
+    }
+  }
+
+  /**
+   * Helper method to create appropriate network error responses for count
+   */
+  private createCountNetworkErrorResponse(error: unknown): EnsRainbow.CountResponse {
+    let errorMessage = "Unknown network error occurred";
+
+    if (error instanceof Error) {
+      errorMessage = error.message;
+
+      // DOMException is thrown for aborts, including timeouts
+      if (error.name === "AbortError") {
+        return {
+          status: StatusCode.Error,
+          error: "Request timed out",
+          errorCode: ErrorCode.TIMEOUT,
+        } as EnsRainbow.CountTimeoutError;
+      }
+
+      // Network connectivity issues
+      if (
+        errorMessage.toLowerCase().includes("network") ||
+        errorMessage.toLowerCase().includes("failed to fetch") ||
+        errorMessage.toLowerCase().includes("fetch failed")
```

Comment:
I hope it will catch these errors: https://github.com/namehash/ensnode/issues/214